### DISCUSSION TITLE
Method to convert wide to long format triangle

### DISCUSSION THREAD

**Discussion by xne0n** [casact/chainladder-python#218] [Q&A] [answered]
Hello,
How do you guys convert your wide triangle (imported from, let's say, an excel file to a pandas dataframe) into the tabular long format required ?
I've looked up on the documentation and I don't find anything...
Thank you !

**Comment by jbogaardt** [comment, marked as answer]
Hi @xne0n ,

Thanks for the question. It's not documented.  We should probably mention something in the docs or figure out how to support wide.  I'd use `pandas.unstack` to get there. Something like this:

```python
import chainladder as cl

# Suppose you have a wide triangle (perhaps copied from Excel clipboard...)
df = cl.load_sample('raa').to_frame()

# You can use pandas unstack to make wide format long
df = df.unstack().dropna().reset_index()
# make sure you have good column labels
df.columns = ['age', 'origin', 'values']
# and make a date-like valuation
df['valuation'] = (df['origin'].dt.year + df['age'] / 12 - 1).astype(int)
# Finally make a Triangle
cl.Triangle(df, origin='origin', development='valuation', columns='values')
```

**Reply by xne0n** [reply]
Thank you !
